### ISSUE TITLE
Front-end Work

### ISSUE THREAD

**Issue by banesullivan** [ResonantGeoData/RGD-ScrumBoard#32]
Target deployment: build a general UI for Resonant GeoData with RD-OpenGeo as the primary project using it 

Take a look at the initial UI mockups for a Single Page App *for inspiration*: https://kwgitlab.kitware.com/ux-design/design-tracker/-/issues/81

- [x] 1. Bootstrap new UI in a new repo under this organizational account
    - left panel for searches
    - large Cesium view on the right with dummy data
    - empty navbar
- [x] 2. Handle user login, basic auth
- [x] 3. Remove VueCesium library and implement vanilla CesiumJS
- [x] 4. Populate out the search form, hit endpoint to fetch search results, and populate a list/table view
    - Create a search form in a panel with the same field as the current search form in the SSR templates
    - Add button to submit the form for the endpoint `/rgd/search` (see https://www.resonantgeodata.com/swagger/)
    - Collect **paginated** results from search endpoint response (list of `SpatialEntry` objects in the `results` field)
    - Populate a table view in a panel next to the search form with the results. For now, just list the ID (until @banesullivan improves the serializer to have more useful info)
- [x] 5. Populate search results in Cesium view
    - For each search result, hit the endpoint `/rgd​/spatial_entry​/{spatial_id}​/footprint` to fetch the GeoJSON footprint for that entry
    - add these GeoJSON polygons to the cesium map view
    - Perhaps this is helpful: https://cesium.com/learn/cesiumjs/ref-doc/GeoJsonDataSource.html
- [x] 6. Click search result on map (or list view) and see pop up with details
    - this will list basic metadata and just give a link to the old server-rendered pages for now

TODO:

- [x] @banesullivan: CORS on RD-OpenGeo (see https://github.com/ResonantGeoData/RD-OpenGeo/pull/15)
- [x] @aashish24: which project to charge?
- [x] @banesullivan: flush out 3,4,5 details/specifics

**Comment by brianhelba** [comment]
Once we have any initial work, we'll create a deployment with Netlify for easy demos.

**Comment by banesullivan** [comment]
@marySalvi, please update the status of this work and draft a checklist to get us to a minimally viable product.

Minimally viable product entails i-iv of the tasks listed in the description of this issue. Please update if there is anything else you think we need to include in the first iteration.

From there, we can work to set a timeline on this work and outline a new issue/set of tasks for the second iteration of this front-end

**Comment by marySalvi** [comment]
@banesullivan  Please note that I have updated to checklist to include removing the VueCesium component library. Once that is done I will be able to continue work on the rest of the tasks and sub-tasks for a minimally viable product (now iii-v)

**Comment by banesullivan** [comment]
I think it's time we close this
